Packs/MicrosoftGraphApplications/Integrations/MicrosoftGraphApplications/MicrosoftGraphApplications.py

```diff
@@ -162,10 +348,84 @@ def remove_service_principals_command(ms_client: Client, args: dict) -> CommandR
     Returns:
         Results to post in demisto
     """
-    app_id = str(args.get('id'))
-    ms_client.delete_service_principals(app_id)
+    id_for_request, service_id = validate_service_principal_input(args=args)
+    ms_client.delete_service_principals(id_for_request)
+    return CommandResults(
+        readable_output=f'Service {service_id} was deleted.'
+    )
+
+
+def add_password_service_principal_command(ms_client: Client, args: dict) -> CommandResults:
+    """
+    Adds a strong password or secret to the service principal.
+
+    Arguments:
+        ms_client: The Client
+        args: demisto.args()
+
+    Returns:
+        Results to post in demisto
+    """
+    id_for_request, service_id = validate_service_principal_input(args=args)
+
+    origin_keys = ["display_name", "end_date_time", "start_date_time"]
+    alternative_names = ["displayName", "endDateTime", "startDateTime"]
+
+    data = {
+        alternative_name: args[origin_key]
+        for origin_key, alternative_name in zip(origin_keys, alternative_names)
+        if origin_key in args
+    }
+
+    results = ms_client.add_password_service_principal(id_for_request, data=data)
+
+    return CommandResults(
+        'MSGraphApplication',
+        'id',
+        outputs=results,
+        readable_output=f'A password was added to application {service_id} successfully.'
+    )
+
+
+def remove_password_service_principal_command(ms_client: Client, args: dict) -> CommandResults:
+    """
+    Remove a password from the service principal.
+
+    Arguments:
+        ms_client: The Client
+        args: demisto.args()
+
+    Returns:
+        Results to post in demisto
+    """
+    id_for_request, service_id = validate_service_principal_input(args=args)
+
+    data = {"keyId": args["key_id"]}
+
+    ms_client.remove_password_service_principal(id_for_request, data=data)
+
+    return CommandResults(
+        readable_output=f'The password of the unique identifier {args["key_id"]} was removed successfully.'
+    )
+
+
+def unlock_configuration_service_principal_command(ms_client: Client, args: dict) -> CommandResults:
+    """
+    Unlock configuration of a service principal.
+
+    Arguments:
+        ms_client: The Client
+        args: demisto.args()
+
+    Returns:
+        Results to post in demisto
+    """
+    object_id = args["id"]  # required
```

Comment:
```suggestion
    object_id = args["id"]
```
this is implied by the use of direct accessor :)